flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HdfsBehaviorTest.java

```diff
@@ -47,19 +43,17 @@ public class HdfsBehaviorTest extends FileSystemBehaviorTestSuite {
 
     // ------------------------------------------------------------------------
 
-    @BeforeClass
-    public static void verifyOS() {
-        Assume.assumeTrue(
-                "HDFS cluster cannot be started on Windows without extensions.",
-                !OperatingSystem.isWindows());
+    @BeforeAll
+    static void verifyOS() {
+        assumeThat(OperatingSystem.isWindows())
+                .describedAs("HDFS cluster cannot be started on Windows without extensions.")
+                .isFalse();
     }
 
-    @BeforeClass
-    public static void createHDFS() throws Exception {
-        final File baseDir = TMP.newFolder();
-
+    @BeforeAll
+    static void createHDFS(@TempDir java.nio.file.Path tmp) throws Exception {
```

Comment:
You could use `File` instead of `java.nio.file.Path`. This way, you avoid using a qualified type here and there's no need to call `toString()` further down in line 56 because there's `tmp.getAbsolutePath()`.

Reply:
I made this change because of the `getAbsolutePath`, but I'd argue that it's strictly not necessary or desireable to avoid fully qualified classes!  Most uses of `@TempDir` are on `java.nio` classes I believe!